lesson14.md

```diff
@@ -0,0 +1,313 @@
+<!-- .slide: id="lesson14" -->
+
+# Basic Frontend - Fall 2021
+
+Lesson 14, Tuesday, 2021-11-09
+
+---
+
+<!-- .slide: id="loops" -->
+
+## Loops
+
+---
+
+### Lesson overview
+
+- examples of loops
+- while loop
+- for loop
+
+---
+
+### Loops
+
+We use loops to run the same code in our programs repeatedly.
+
+More generally, a loop is a way to repeat an action multiple times as long as a certain condition is `true`.
+
+---
+
+### Loops: cake recipe
+
+Let's say I want to bake a cake:
+
+<div style="display: flex;">
+
+![scale](images/lesson-14-scale.webp) <!-- .element: style="width: 400px"-->
+
+<div>
+Cake recipe:<br>
+
+- 6 eggs
+- 3 cups of sugar
+- 5 cups of flour
+</div>
+</div>
+
+---
+
+### Loops: cake recipe
+
+We take an empty bowl and keep our ingredients ready.
+
+We need 5 cups of flour.
+
+1. What's the action that we need to repeat?
+1. For how long do we need to repeat it?
+
+---
+
+### Loops: cake recipe
+
+1. The action we need to repeat is adding a cup of flour to the bowl
+1. We repeat it 5 times
+
+---
+
+### Loops: cake recipe
+
+We can distinguish different parts in a loop:
+
+- initialization: a starting point
+- condition: a test to see if the loop should stop
+- update: changes the value checked by the condition
+- action: what we want to repeat
+
+---
+
+### Loops: cake recipe
+
+- initialization: no flour in the bowl (`cupsOfFlour = 0`)
+- condition: did we put less than 5 cups of flour? (`cupsOfFlour < 5`)
+- update: cups of flour in the bowl (`cupsOfFlour++`)
+- action: add flour to the bowl
+
+---
+
+### While loop
+
+Let's see how to make an actual loop in JavaScript.
+
+A `while` loop repeats its body while its `condition` is `true`:
+
+```js
+while (condition) {
+  // body is executed while "condition" is true
+  console.log("Hello from while loop");
+}
+```
+
+---
+
+### While loop
+
+```js
+let cupsOfFlour = 0; // initialization
+while (cupsOfFlour < 5) { // condition
+    console.log("Take a cup of flour and put it in the bowl");
+    cupsOfFlour++; // update
+    console.log("There are now " + cupsOfFlour + " cups of flour in the bowl");
+}
+```
+
+---
+
+### While loop
+
+```js
+let bitesOfChocolate = 0;
+
+while (bitesOfChocolate < 10) {
+  bitesOfChocolate++;
+  console.log("Bites of chocolate: " + bitesOfChocolate);
+  console.log("Still hungry.");
+}
+```
+
+As long as we have had less than 10 bites of chocolate, we will have one more and console.log that we're still hungry.
+
```

Comment:
```suggestion
As long as we have had less than 10 bites of chocolate, we will have one more bite and we will still be hungry.
```
Maybe also replace "had" with eaten, used or consumed to illustrate that "you are still hungry until you eat all the bites needed to not feel hungry anymore"? Also, do we want to log "Still hungry" after the last bite? I assume that after the last bite, you're no longer hungry. It might be a bit confusing to add a condition for this case though.